tests/e2e/blocks/carousel-header.spec.js

```diff
@@ -1,67 +1,67 @@
 import {test, expect} from '../tools/lib/test-utils.js';
 import {publishPostAndVisit, createPostWithFeaturedImage} from '../tools/lib/post.js';
+import {searchAndInsertBlock} from '../tools/lib/editor.js';
 
 test.useAdminLoggedIn();
 
 test('Create and check carousel header block', async ({page, admin, editor}) => {
-  await createPostWithFeaturedImage({admin, editor}, {title: 'Test Carousel', postType: 'page'});
+  await createPostWithFeaturedImage({page, admin, editor}, {title: 'Test Carousel', postType: 'page'});
 
-  await editor.canvas.getByRole('button', {name: 'Add default block'}).click();
-  await page.keyboard.type('/carousel');
-  await page.getByRole('option', {name: 'Carousel Header'}).click();
+  // Add block
+  await searchAndInsertBlock({page}, 'Carousel Header');
 
   // Filling Carousel details
-  const title = page.getByRole('textbox', {name: 'Enter title'});
-  const description = page.getByRole('textbox', {name: 'Enter description'});
-  const ctaBtn = page.getByRole('textbox', {name: 'Enter CTA text'});
-
+  let title = page.getByRole('textbox', {name: 'Enter title'});
   await title.click();
   await title.fill('My test Header 1');
 
+  let description = page.getByRole('textbox', {name: 'Enter description'});
   await description.click();
   await description.fill('Testing carousel description 1');
 
+  let ctaBtn = page.getByRole('textbox', {name: 'Enter CTA text'});
   await ctaBtn.click();
   await ctaBtn.fill('Read more 1');
 
-  // Check if sidebar is not visible
+  // // Check if sidebar is not visible
   await editor.openDocumentSettingsSidebar();
 
-  await page.click('[data-type="toggle"]');
-  await page.getByLabel('Url for link').fill('https://google.com');
+  await page.locator('[data-type="toggle"]').nth(0).click();
+  await page.getByLabel('Url for link').nth(0).fill('https://google.com');
 
   await page.getByRole('button', {name: 'Edit'}).click();
   await page.getByRole('button', {name: 'Add image'}).click();
   await page.getByRole('tab', {name: 'Media Library'}).click();
-  await page.getByRole('checkbox', {name: 'OCEANS-GP0STOM6C'}).click();
+  await page.getByRole('checkbox', {name: 'NATURE-GP0STOE2U'}).click();
   await page.getByRole('button', {name: 'Select', exact: true}).click();
 
   // Next Slide
   await page.getByRole('button', {name: 'Edit'}).click();
   await page.getByRole('button', {name: 'Add slide'}).click();
 
-  const title2 = page.getByRole('textbox', {name: 'Enter title'}).nth(1);
-  const description2 = page.getByRole('textbox', {name: 'Enter description'}).nth(1);
-  const ctaBtn2 = page.getByRole('textbox', {name: 'Enter CTA text'}).nth(1);
-
-  await title2.click();
-  await title2.fill('My test Header 2');
+  // Re-declare variables
+  title = page.getByRole('textbox', {name: 'Enter title'}).nth(1);
```

Comment:
I'm wondering if there is a way to remove some code duplication here with a loop or something like that 🤔

Reply:
Just updated. Please check if you agree !